### ISSUE TITLE
Clarify AZ anti-affinity rules for CREATE CLUSTER REPLICA

### ISSUE THREAD

**Issue by jseldess** [MaterializeInc/materialize#18584]
### Documentation request

If we have anti-affinity rules that place replicas in different AZ, we should clarify that in the `CREATE CLUSTER` and `CREATE CLUSTER REPLICAS` docs and perhaps other docs.

If we don't have such anti-affinity rules, we should tweak the `CREATE CLUSTER` example queries to use different availability zones when creating multi-replica clusters:

```
CREATE CLUSTER c1 REPLICAS (
    r1 (SIZE = 'medium', AVAILABILITY ZONE = 'use1-az1'),
    r2 (SIZE = 'medium', AVAILABILITY ZONE = 'use1-az2')
);
```

From thread with @aalexandrov: https://materializeinc.slack.com/archives/CPNFV9CVB/p1680613760992589

**Comment by jseldess** [comment]
Slack thread revealed answers: We do spread replicas across AZs automatically when AZs aren't specified. But in this case, we should note that replicas can change AZs between restarts, which happens for every upgrade, at this time.

**Comment by benesch** [comment]
>  But in this case, we should note that replicas can change AZs between restarts, which happens for every upgrade, at this time.

Mentioned this on Slack just now, but I don't think this is true! At least that's not what my read of the code suggests. Someone should double check, though, by creating a replica and restarting envd a bunch and making sure it keeps its original AZ after each restart.

If this isn't true, it's a bug, and we should fix it ASAP as it impacts the availability of the system if envd crashes.

**Comment by morsapaes** [comment]
Closed in #18618.
